typing/jkind.ml

```diff
@@ -1845,6 +1936,11 @@ let for_object =
 (******************************)
 (* elimination and defaulting *)
 
+let[@inline] normalize ~require_best ~jkind_of_type t =
+  { (disallow_right t) with
+    jkind = Jkind_desc.normalize ~require_best ~jkind_of_type t.jkind
```

Comment:
If `not require_best`, then the updated jkind should be `not_best` I think

Reply:
I'd like @liam923's take on this, since he shifts this stuff around a bit in the subsequent PR

Reply:
My pr makes it so that if it runs out of fuel then it gets `Not_best`. But it doesn't do what Richard suggests, and I think that'd be a good change?

Reply:
Do you want to make that change on your branch, or should I?

Reply:
I've just made this change on this branch and pushed